llvm/lib/Target/PowerPC/PPCISelDAGToDAG.cpp

```diff
@@ -7578,6 +7668,10 @@ void PPCDAGToDAGISel::PeepholePPC64() {
     if (isVSXSwap(SDValue(N, 0)))
       reduceVSXSwap(N, CurDAG);
 
+    // This optimization is performed for non-TOC-based local-exec accesses.
+    if (HasAIXSmallLocalExecTLS)
+      foldADDIForLocalExecAccesses(N, CurDAG);
```

Comment:
if  foldADDIForLocalExecAccesses success ? do we still want to go down the code ?

Reply:
I think this is fine. Since we can fold addi instructions first, and then after there may be load and store instructions that we can fold, which can only happen if we continue further into this loop.

Reply:
if foldADDIForLocalExecAccesses success only when N->getMachineOpcode() == PPC::ADDI8. and the Opcode of N is not changed in the foldADDIForLocalExecAccesses  , so the OpeCode of N is still ADDI8

it will be `continue` here. 
```
unsigned FirstOp;
    unsigned StorageOpcode = N->getMachineOpcode();
    bool RequiresMod4Offset = false;
 switch (StorageOpcode) {
    default: continue;
```

is it correct ?

Reply:
I think I see what you mean. If we have successfully did the transformation in `foldADDIForLocalExecAccesses()`, we're still going to have `PPC::ADDI8` and it will continue down into the code below.

I think this is fine and is what I initially expected, because as you mentioned, it will hit the `default: continue;` code, which makes sense all of the code below that point checks for if `N` is a load/store (which will be false since we have an `PPC::ADDI8` in the current iteration), so it will go into checking the next `N` (which could be an addi, load, store, etc) and do the most suitable transformation depending on the opcode.

I think what I have done here would be similar to the check and transformation above my change,
```
    if (isVSXSwap(SDValue(N, 0)))
      reduceVSXSwap(N, CurDAG);
```

Reply:
Thanks for the suggestion, Digger. I was initially thinking of keeping the code as is.

However, just so I understand your suggestion, are you suggesting the following or am I mistaken?
```
    if (HasAIXSmallLocalExecTLS) {
      foldADDIForLocalExecAccesses(N, CurDAG);
      continue;
    }
```

In the case where I have something like:
```
SelectionDAG has 9 nodes:
  t0: ch,glue = EntryToken
      t2: i64,ch = CopyFromReg t0, Register:i64 %0
      t22: i64 = ADDI8 Register:i64 $x13, TargetGlobalTLSAddress:i64<ptr @IThreadLocalVarInit> 0 [TF=7]
    t17: ch = STB8<Mem:(store (s8) into %ir.0)> t2, TargetConstant:i64<0>, t22, t0
  t13: ch = BLR8 t17
```
It will visit the `STB8` first, and it will fail the transformation for `foldADDIForLocalExecAccesses()`, but since I have a `continue;` right after, it will not proceed below to optimize the store.

Just wanted to double check if I understood your suggestion correctly, because I think currently as I understand it, it doesn't seem like it would work in this situation.